src/main/kotlin/uk/gov/justice/digital/hmpps/incentivesapi/resource/IncentiveLevelResource.kt

```diff
@@ -311,6 +323,6 @@ class IncentiveLevelResource(
     @PathVariable
     code: String,
   ): IncentiveLevel {
-    return partiallyUpdateIncentiveLevel(code, IncentiveLevelUpdate(active = false))
```

Comment:
We're not checking for `required` here to validate the request. We could.

I'm guessing we're trusting the user to know what they're doing and avoiding doing this in 2 steps?:
- mark a level as optional
- then deactivate a level

Reply:
i guess it needs better explanation in the swagger docs at least.
we still don't really know what happens when a level is deactivated globally (as yet, nothing). 🤷

Reply:
True.

Part of me thinks this is simpler but the other part wonders whether not having the validation means it's easy to accidentally deactivate a _required_ level (as it would set `required=false` behind the scenes for the user/client)

Reply:
well the user here is just our team cos i don't see why anybody else would want to delete levels ;)
i'll look at the swagger docs at least, but can't decide whether patch + delete is actually better. i mean, the delete action is really dangerous irrespective of whether a level is required.